### PR TITLE
V2 attached connector

### PR DESCRIPTION
**Pull request by grs**

### CODE REVIEW

**Comment by grs** [comment]
This allows a site to target pods in other namespaces. It adds two new resource types. The AttachedConnector resource is declared in the namespace where the pods to be tracked are running. It is largely similar to the Connector resource,  but has a siteNamespace field that identifies the site in which it is desired that the pods being tracked are used to configure connectors. The other new resource type is the AttachedConnectorAnchor. This is created in the namespace of the site for which an AttachedConnector is to be defined.

To track pods in another namespace, a pair of resources, one of each of these new types is required. These should use the same resource name. The AttachedConnector is defined in the namespace where the pods to be tracked are running. The AttachedConnectorAnchor is defined in the site that should be targetting those pods. This way, RBAC can be used to ensure that appropriate access to each namespace is required (not necessarily by the same person) in order for a site to be able to track pods in another namespace. Each of the resources in the pair references the namespace of the other. In the case of the AtttachedConnector, this is done through the siteConnector field. In the case of the AttachedConnectorAnchor, this is done through the connectorNamespace field.

File: api/types/crds/skupper_attached_connector_anchor_crd.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        type: boolean
+        description: Whether connector is active
+        jsonPath: .status.active
+      - name: Status
+        type: string
+        description: The status of the connector
+        jsonPath: .status.status
+  scope: Namespaced
+  names:
+    plural: attachedconnectoranchors
```
Comment (fgiorgetti) [inline api/types/crds/skupper_attached_connector_anchor_crd.yaml:52]: Just a suggestion. Maybe adding a shortName, like: `aca` or something similar.

File: api/types/crds/skupper_attached_connector_crd.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+        type: boolean
+        description: Whether connector is active
+        jsonPath: .status.active
+      - name: Status
+        type: string
+        description: The status of the connector
+        jsonPath: .status.status
+  scope: Namespaced
+  names:
+    plural: attachedconnectors
```
Comment (fgiorgetti) [inline api/types/crds/skupper_attached_connector_crd.yaml:65]: Maybe a shortName would be good as well?

**Review by fgiorgetti — APPROVED** [review]
A minor issue only and a few comments, but overall it looks good.
I like the attention you gave to the separation of attached connector and anchor,
to ensure RBAC will properly support exposing resources from other namespaces.
Great work, as usual.

File: api/types/crds/skupper_attached_connector_anchor_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+          properties:
+            spec:
+              type: object
+              properties:
+                connectorNamespace:
+                  type: string
+                routingKey:
+                  type: string
+              required:
+              - routingKey
```
[the remark this answers is not in the document]
Comment (grs) [reply api/types/crds/skupper_attached_connector_anchor_crd.yaml:23]: Yes, good point, it should be required.

File: api/types/crds/skupper_attached_connector_crd.yaml [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      served: true
+      storage: true
+      schema:
+        openAPIV3Schema:
+          type: object
+          properties:
+            spec:
+              type: object
+              properties:
+                siteNamespace:
```
[the remark this answers is not in the document]
Comment (grs) [reply api/types/crds/skupper_attached_connector_crd.yaml:18]: Agreed.

File: api/types/crds/skupper_attached_connector_anchor_crd.yaml [hunk 5]
```diff
@@ -0,0 +1,54 @@
+apiVersion: apiextensions.k8s.io/v1
+kind: CustomResourceDefinition
+metadata:
+  name: attachedconnectoranchors.skupper.io
```
[the remark this answers is not in the document]
Comment (grs) [reply api/types/crds/skupper_attached_connector_anchor_crd.yaml:4]: Oops, yes, will do.

File: pkg/kube/site/attached_connector.go [hunk 6]
```diff
@@ -0,0 +1,362 @@
...
+}
+
+func (b *ExtendedBindings) checkAttachedConnectorAnchor(namespace string, name string, anchor *skupperv1alpha1.AttachedConnectorAnchor) error {
+	connector, ok := b.connectors[name]
+	if !ok {
+		connector = NewAttachedConnector(name, namespace, b)
+		b.connectors[name] = connector
+	}
+	if (anchor == nil && connector.anchorDeleted()) || (anchor !=nil && connector.anchorUpdated(anchor)) {
+		if err := b.site.updateRouterConfigForGroups(b.site.bindings); err != nil {
```
[the remark this answers is not in the document]
Comment (grs) [reply pkg/kube/site/attached_connector.go:73]: Agreed. Will check for that.

**Review by pwright — COMMENTED** [review]
Should descriptions be sentences? ie Start with capital letter and finish with a full stop?

**Review by nluaces — APPROVED** [review]

**Review by ajssmith — APPROVED** [review]
lgtm

File: api/types/crds/skupper_attached_connector_anchor_crd.yaml (hunk 1 above)
Comment (grs) [reply api/types/crds/skupper_attached_connector_anchor_crd.yaml:89]: I agree that short names would be useful. I feel aca might be a little too opaque (though cm, ss are now well understood). I would like to address short names across all the CRDs in a separate PR.

**Pull request merged**